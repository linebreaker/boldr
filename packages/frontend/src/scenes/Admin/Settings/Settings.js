/* @flow */
import React, { PureComponent } from 'react';
import Helmet from 'react-helmet';
import styled from 'styled-components';
import { Grid, Col, Row } from '@boldr/ui/Layout';
import Headline from '@boldr/ui/Headline';
import Paper from '@boldr/ui/Paper';

import {
  Accordion,
  AccordionItem,
  AccordionItemTitle,
  AccordionItemBody,
} from '@boldr/ui/Accordion';
// import { HeaderPanel, SettingsPanel } from 'boldr-ui';
import SettingsAside from './components/SettingsAside';
import General from './components/General';

type Props = {
  settings: Array<Setting>,
};
function filterSiteName(obj) {
  return parseInt(obj.id, 10) === 1;
}
function filterSiteUrl(obj) {
  return parseInt(obj.id, 10) === 2;
}
function filterSiteLogo(obj) {
  return parseInt(obj.id, 10) === 3;
}
function filterSiteDescript(obj) {
  return parseInt(obj.id, 10) === 4;
}
function filterSiteFav(obj) {
  return parseInt(obj.id, 10) === 5;
}

const Content = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  padding: 0;
  height: 100%;
`;
class Settings extends PureComponent {
  props: Props;

  render() {
    const { settings } = this.props;
    const siteName = settings.find(filterSiteName);
    const siteLogo = settings.find(filterSiteLogo);
    const siteDescription = settings.find(filterSiteDescript);
    const siteFavicon = settings.find(filterSiteFav);
    const siteUrl = settings.find(filterSiteUrl);
    return (
      <div>
        <Helmet title="Admin: Settings" />
        <Content>
          <SettingsAside />
          <General />
        </Content>
      </div>
    );
  }
}

export default Settings;
