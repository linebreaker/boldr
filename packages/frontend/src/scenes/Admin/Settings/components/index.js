import Favicon from './Favicon';
import Logo from './Logo';
import SiteDescription from './SiteDescription';
import SiteName from './SiteName';
import SiteUrl from './SiteUrl';

export { Favicon, Logo, SiteDescription, SiteName, SiteUrl };
