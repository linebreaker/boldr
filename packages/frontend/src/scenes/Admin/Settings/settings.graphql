query GetSettingsQuery {
  settings {
    id
    key
    value
    label
    description
  }
}
