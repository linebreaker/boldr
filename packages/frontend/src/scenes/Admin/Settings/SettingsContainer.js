/* @flow */
import React from 'react';
import { graphql } from 'react-apollo';
// internal
import Loader from '@boldr/ui/Loader';
import Settings from './Settings';
import SETTINGS_QUERY from './settings.graphql';

type Props = {
  data: Data,
};
interface Data {
  settings: Array<Setting>,
  loading: boolean,
}

const SettingsContainer = (props: Props) => {
  const { loading, settings } = props.data;
  if (loading) {
    return <Loader />;
  }
  return <Settings settings={settings} />;
};

export default graphql(SETTINGS_QUERY, {
  options: () => ({
    fetchPolicy: 'cache-and-network',
  }),
})(SettingsContainer);
