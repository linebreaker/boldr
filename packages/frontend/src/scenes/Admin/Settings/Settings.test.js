import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import fakeSettings from './__fixtures__/settings.fixture';
import Settings from './Settings';

describe('<Settings />', () => {
  const wrapper = shallow(<Settings settings={fakeSettings} />);
  it('renders <Settings /> without breaking', () => {
    expect(wrapper.find('div').length).toBe(1);
  });
  it('renders snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
