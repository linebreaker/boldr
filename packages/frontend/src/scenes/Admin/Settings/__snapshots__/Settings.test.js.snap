// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Settings /> renders snapshot 1`] = `
<div>
  <HelmetWrapper
    encodeSpecialCharacters={true}
    title="Admin: Settings"
  />
  <Settings__Content>
    <SettingsAside />
    <Connect(General) />
  </Settings__Content>
</div>
`;
